eras/byron/ledger/impl/src/Cardano/Chain/Update/Validation/Endorsement.hs

```diff
@@ -172,8 +172,7 @@ register env st endorsement =
     isConfirmedAndStable upId = upId `M.member` scps
       where
         -- Stable and confirmed proposals.
-        scps = M.filter (<= stableAt) confirmedProposals
-        stableAt = subSlotCount (kSlotSecurityParam k) currentSlot
+        scps = M.filter (\x -> addSlotCount (kSlotSecurityParam k) x <= currentSlot) confirmedProposals
```

Comment:
I don't know what the `confirmedProposals` is, but if they contain untrusted input then there is a danger of overflow, which before was guarded against with `subSlotCount`. Example:
```haskell
λ> let currentSlot = SlotNumber 10
λ> let k = SlotCount 5
λ> filter (\x -> x <= subSlotCount k currentSlot) (map SlotNumber [1,2,30, maxBound])
[SlotNumber {unSlotNumber = 1},SlotNumber {unSlotNumber = 2}]
λ> filter (\x -> addSlotCount k x <= currentSlot) (map SlotNumber [1,2,30, maxBound])
[SlotNumber {unSlotNumber = 1},SlotNumber {unSlotNumber = 2},SlotNumber {unSlotNumber = 18446744073709551615}]
```

So, I'd suggest we also fix `addSlotCount` just in case:

```haskell
addSlotCount :: SlotCount -> SlotNumber -> SlotNumber
addSlotCount (SlotCount a) (SlotNumber b)
  | a <= maxBound - b = SlotNumber $ a + b
  | otherwise = SlotNumber maxBound 
```

Reply:
it is not untrusted input, `currentSlot` is the current slot number, and `k` is fixed. plus this is Byron code that will never be run on new inputs on mainnet/testnet. But you code is indeed nicer in general. I'm happy to make the change.

Reply:
> it is not untrusted input, currentSlot is the current slot number, and k is fixed.

And what about `confirmedProposals`?

> this is Byron code that will never be run on new inputs 

If this code doesn't run during syncing then it really doesn't matter.

Reply:
yea, `confirmedProposals` is both bounded and only ever supplied by the genesis nodes.

> If this code doesn't run during syncing then it really doesn't matter.
it does runs during sync.so if there was an overflow lurking, a new node would not be able to sync. but there's no worry of a node that has synced of crashing from this.